ckan/lib/munge.py

```diff
@@ -119,6 +127,34 @@ def munge_filename(filename):
     return filename
 
 
+def munge_filename(filename):
+    ''' Tidies a filename
+
+    Keeps the filename extension.
+    Strips off any path on the front.
+    It can be useful to keep the extension (e.g. .csv) in a resource when the
+    format field set.
+    '''
+
+    # just get the filename ignore the path
+    path, filename = os.path.split(filename)
+    # clean up
+    filename = substitute_ascii_equivalents(filename)
+    filename = filename.lower().strip()
+    filename = re.sub(r'[^a-zA-Z0-9. -]', '', filename).replace(' ', '-')
+    # resize if needed but keep extension
+    name, ext = os.path.splitext(filename)
+    # limit overly long extensions
+    if len(ext) > 21:
+        ext = ext[:21]
+    # max/min size
+    ext_length = len(ext)
+    name = _munge_to_length(name, 3 - ext_length, 100 - ext_length)
```

Comment:
This seems wrong `3 - ext_length` can be `-19` a min length of `-19` seems illogical.
`'_' * -19` in `_munge_to_length` will create odd names like `.ext` where as `_.ext` might be better